universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -126,18 +127,40 @@ public void archiveQuestion(Long id) throws InvalidUpdateException {
   }
 
   @Override
-  public void discardDraft(Long id) throws InvalidUpdateException {
-    Optional<Question> question =
-        questionRepository.lookupQuestion(id).toCompletableFuture().join();
-    if (question.isEmpty()) {
-      throw new InvalidUpdateException("Did not find question.");
-    }
+  public void discardDraft(Long draftId) throws InvalidUpdateException {
+    Question question =
+        questionRepository
+            .lookupQuestion(draftId)
+            .toCompletableFuture()
+            .join()
+            .orElseThrow(() -> new InvalidUpdateException("Did not find question."));
+
+    // Find the Active version.
+    Version activeVersion = versionRepositoryProvider.get().getActiveVersion();
+    Long activeId =
+        activeVersion
+            .getQuestionByName(question.getQuestionDefinition().getName())
+            // TODO: If nothing depends on this question then it could be removed.
+            .orElseThrow(() -> new InvalidUpdateException("Deleting the first version of a question is not supported."))
```

Comment:
FYI: I added this too. In the current code if you delete the first revision of a question and a draft program uses it, then the program becomes corrupt as there is not active revision of the program to use.

We could detect and not allow for it in the UI, but not a concern for this PR, and the Program won't be corrupted at least.